Tests/SentryTests/Integrations/SentryCrash/SentryCrashReportTests.swift

```diff
@@ -65,6 +66,51 @@ class SentryCrashReportTests: XCTestCase {
         }
     }
     
+    func testShouldWriteReason_WhenWritingNSException() {
+        var monitorContext = SentryCrash_MonitorContext()
+        
+        let reason = "Something bad happened"
+        reason.withCString {
+            monitorContext.crashReason = $0
+            monitorContext.crashType = SentryCrashMonitorTypeNSException
+            
+            let api = sentrycrashcm_system_getAPI()
+            api?.pointee.addContextualInfoToEvent(&monitorContext)
+            sentrycrashreport_writeStandardReport(&monitorContext, fixture.reportPath)
+        }
+        
+        let crashReportContents = FileManager.default.contents(atPath: fixture.reportPath) ?? Data()
+        do {
+            let crashReport: CrashReport = try JSONDecoder().decode(CrashReport.self, from: crashReportContents)
+            
+            expect(crashReport.crash.error.type) == "nsexception"
+            expect(crashReport.crash.error.reason) == reason
+            expect(crashReport.crash.error.nsexception?.reason) == reason
+        } catch {
+            XCTFail("Couldn't decode crash report: \(error)")
+        }
```

Comment:
l: Suggestion to  quiet  Codecov
```suggestion
        let crashReportContents = FileManager.default.contents(atPath: fixture.reportPath) ?? Data()

        let crashReport: CrashReport = try? JSONDecoder().decode(CrashReport.self, from: crashReportContents)
            
        expect(crashReport?.crash.error.type) == "nsexception"
        expect(crashReport?.crash.error.reason) == reason
        expect(crashReport?.crash.error.nsexception?.reason) == reason
```